### PR TITLE
Docs: Document how to run queries in parallel

### PR DESCRIPTION
**Pull request by andresmgot**
**What this PR does / why we need it**:

Documents https://github.com/grafana/grafana-plugin-sdk-go/pull/988

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-plugin-sdk-go/issues/972

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/tutorials/build-a-data-source-backend-plugin.md [hunk 1]
```diff
@@ -179,6 +179,30 @@ The following instructions are based on Grafana v10.1.1, consult the [documentat
 1. Save the dashboard.
 1. After some time the alert rule evaluates and transitions into _Alerting_ state.
 
+## Run multiple queries concurrently
```
Comment (andresmgot) [inline docusaurus/docs/tutorials/build-a-data-source-backend-plugin.md:182]: why is this all red ? :red_square: :sweat_smile:

Comment (josmperez) [reply docusaurus/docs/tutorials/build-a-data-source-backend-plugin.md:182]: no idea. there doesn't seem to be a coding error.

File: docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md [hunk 2]
```diff
@@ -64,6 +64,7 @@ Is something missing from this list? [Let us know](https://github.com/grafana/pl
 - **Plugins should not execute arbitrary code in the backend** - Again, this is a security risk and should be avoided. If your plugin needs to execute code, provide a list of allowed commands and validate the input before executing it.
 - **In general, any error happening should be logged with level `error`.**
 - **Don't use the `info` level: use the `debug` level instead.**
+- **Enable concurrent query execution** - This allows to run multiple queries in parallel. To learn more, refer to our [documentation](/tutorials/build-a-data-source-backend-plugin#run-multiple-queries-concurrently).
```
Comment (josmperez) [inline docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md:67]:
```suggestion
- **Enable concurrent query execution** - This allows you to run multiple queries in parallel. To learn more, refer to our [documentation](/tutorials/build-a-data-source-backend-plugin#run-multiple-queries-concurrently).
```

**Review by josmperez — APPROVED** [review]
A few suggestions and a commit with minor edits. LGTM.

**Pull request merged**